### DISCUSSION TITLE
Prevent changes to the base environment by non-root users

### DISCUSSION THREAD

**Discussion by qomhmd** [conda-forge/miniforge#624] [Ideas]
We have some servers with previous multi-user Conda instances on it. I started to use Miniforge on the newer ones and want to prevent users from cluttering up the base environment. It seems there is no implemented configuration to do such a thing. I did not find envs_dirs and pkgs_dirs useful for this purpose. 

Moreover, I tried adding this code to a bashrc profile, but only the pip part is working. It seems I should deal with dedicated miniforge3/etc/profile.d/conda.sh or so to make it happen.

```shell
function get_conda_env() {
  current_env=$(conda info --envs | grep '*' | awk '{print $1}')
  echo "$current_env"
}

function pip_unbased() {
  current_env=$(get_conda_env)

  if [ "$current_env" = "base" ] && [[ "$1" = "install" || "$1" = "uninstall" ]]; then
    echo "Warning: Changing Conda's base environment is not allowed. You should create and/or activate another environment."
  else
    command pip "$@"
  fi
}

function conda_unbased() {
  current_env=$(get_conda_env)

  if [[ "$current_env" = "base" || "$current_env" = "" ]] && [[ "$1" = "install" || "$1" = "remove" ]]; then
    echo "Warning: Changing Conda's base environment is not allowed. You should create and/or activate another environment."
  else
    command conda "$@"
  fi
}

alias conda=conda_unbased
alias mamba=conda_unbased
alias pip=pip_unbased

```

**Comment by jaimergp** [comment]
`conda` is already a shell function so maybe that's why it doesn't work.

I wonder if you can handle this with your OS permissions, though. Make `base` readable and executable, but not writable. Maybe do allow write in `pkgs/` (and `envs/`?) so the cache doesn't get duplicated across user homes.

**Comment by pyinstallerbaggins** [comment]
Have you looked into Jupyterhub for its permissions and access capabilities? 

Try out Nebari or mybinder.org? 

Let me know what you think!

**Comment by gernophil** [comment]
I am looking for the same thing. Somehow overlooked yours yesterday when I opened [my thread](https://github.com/conda-forge/miniforge/discussions/663). Seems like you've adapted the code from [here](https://github.com/conda/conda/issues/7791#issuecomment-945676641). However, they are using `$CONDA_PROMPT_MODIFIER`. Could that work?

EDIT: they also don't use `command conda "$@"`, but only `conda "$@"`.
